Kinvey/Kinvey/RealmSync.swift

```diff
@@ -56,6 +56,12 @@ class RealmSync<T: Persistable>: SyncType where T: NSObject {
         log.verbose("Saving pending operation: \(pendingOperation)")
         executor.executeAndWait {
             try! self.realm.write {
+                if !pendingOperation.collectionName.isEmpty,
```

Comment:
the entire queue, but Realm does not load the entire result set in memory. it does incrementally.